### ISSUE TITLE
HubSpot integration not mapping boolean values correctly

### ISSUE THREAD

**Issue by Mosnar** [verbb/formie#472]
**Description**

Currently the HubSpot integration is passing boolean fields as boolean values; however, HubSpot requires these values to be strings (for some reason). 

I've tried it with `"Yes"` and `"No"` and that has worked; however, I believe you can also pass `"true"` and `"false"` as strings rather than bools. Currently, the value is passed as `0` or `1` and this results in an API error.

queue.log excerpt:
```
2021-05-19 16:17:58 [-][-][-][error][formie] Hubspot: API error: “{"validationResults":[{"isValid":false,"message":"1 was not one of the allowed options: [label: \"Yes\"\nvalue: \"true\"\ndisplay
_order: 0\nhidden: false\nread_only: false\n, label: \"No\"\nvalue: \"false\"\ndisplay_order: 1\nhidden: false\nread_only: false\
n]","error":"INVALID_OPTION","name":"newsletter_opt_in"}],"status":"error","message":"Property values were not valid","correlationId":"5b295cdd-4932-4020-843b-f92b3b46ced2"}” /vendor/guzzlehttp/guzzle/src/Exception/RequestException.php:113
```

Here's the excerpt from the documentation:
```
textarea - a <textarea> field, stores data as a string
text - a simple text box, stores a string
date - a datepicker field, stores a date type
file - stores the URL location of a file. Note: The file itself must be stored separately, as the contact property cannot store the file, just the URL location of a file. Treated as a string.
number - a number input field, stores a number value
select - a dropdown box, uses the enumeration type
radio - a set of radio buttons, used with the enumeration data type.
checkbox - a set of checkboxes, used with the enumeration data type
booleancheckbox - a single checkbox, stores "true" (as a string) if checked.
```
Reference: https://legacydocs.hubspot.com/docs/methods/contacts/v2/create_contacts_property

**Additional info**
- Plugin version: 1.4.4
- Craft version: 3.6.15

**Comment by engram-design** [comment]
Interesting. We're actually using the v3 API, but this might be applicable in either case if it's not working at the moment.

It should be a case of removing this line:

https://github.com/verbb/formie/blob/e985eee6d923bf27dc4f98e88265e454182ccdee/src/integrations/crm/HubSpot.php#L403

As the default variable type will be a string.

**Comment by Mosnar** [comment]
I updated the original issue with the log excerpt. Here's the work-around I ended up using that solved it:

```php
        Event::on(Submission::class, Submission::EVENT_MODIFY_FIELD_VALUE_FOR_INTEGRATION, function(ModifyFieldValueForIntegrationEvent $event) {
            $field = $event->field;
            if ($field instanceof Agree) {
                if ($event->value === true) {
                    $event->value = 'Yes';
                } else {
                    $event->value = 'No';
                }
            }
        });
```

**Comment by engram-design** [comment]
Ah, good point, setting it to a string won't be enough. I believe I have a similar handling of this somewhere, so I'll bare that in mind for Hubspot!

**Comment by engram-design** [comment]
I also stand corrected on using the v1 API for the contact endpoint. HubSpot's API's are a bit all over the show. Fixed for the next release. To get the fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.4.4",
  "...": "..."
}
```

Then run `composer update`. You'll also have to refresh the Integration, then save the form in order to get these new types for fields.

**Comment by engram-design** [comment]
Fixed in [1.4.5](https://github.com/verbb/formie/releases/tag/1.4.5)
